src/urllib3/contrib/pyopenssl.py

```diff
@@ -372,14 +372,15 @@ def shutdown(self):
         self.connection.shutdown()
 
     def close(self):
-        if self._makefile_refs < 1:
-            try:
-                self._closed = True
-                return self.connection.close()
-            except OpenSSL.SSL.Error:
-                return
-        else:
-            self._makefile_refs -= 1
```

Comment:
ah this is a pain, pypy2.7 socket.close is broken in the same way as this so this will need to be conditional based on pypy2.7